auto_submit/test/service/validation_service_test.dart

```diff
@@ -674,4 +674,226 @@ void main() {
+      commentList.add(issueComment);
+      final MergeMethod mergeMethod = await validationService.checkForMergeMethod(
+        slug: slug,
+        prNumber: 1,
+      );
+      expect(MergeMethod.rebase, mergeMethod);
+    });
+
+    test('Multiple lines in single comment return last found merge method.', () async {
+      const String issueCommentBody = '@autosubmit:rebase\n@autosubmit:merge';
+      final IssueComment issueComment = IssueComment(
+        id: 1,
+        body: issueCommentBody,
+        authorAssociation: 'OWNER',
+      );
+      commentList.add(issueComment);
+      final MergeMethod mergeMethod = await validationService.checkForMergeMethod(
+        slug: slug,
+        prNumber: 1,
+      );
+      expect(MergeMethod.merge, mergeMethod);
+    });
+
+    test('Multiple comments return last found merge method.', () async {
+      const String issueCommentBody = '@autosubmit:rebase';
+      final IssueComment issueComment = IssueComment(
+        id: 1,
+        body: issueCommentBody,
+        authorAssociation: 'OWNER',
+      );
+      const String issueCommentBody2 = '@autosubmit:merge';
+      final IssueComment issueComment2 = IssueComment(
+        id: 2,
+        body: issueCommentBody2,
+        authorAssociation: 'OWNER',
+      );
+      commentList.add(issueComment);
+      commentList.add(issueComment2);
+      final MergeMethod mergeMethod = await validationService.checkForMergeMethod(
+        slug: slug,
+        prNumber: 1,
+      );
+      expect(MergeMethod.merge, mergeMethod);
+    });
+
+    test('Multiple comments, multiple lines in comment return last found merge method.', () async {
+      const String issueCommentBody = '@autosubmit:rebase\n@autosubmit:merge';
+      final IssueComment issueComment = IssueComment(
+        id: 1,
+        body: issueCommentBody,
+        authorAssociation: 'MEMBER',
+      );
+      const String issueCommentBody2 = '@autosubmit:squash\n@autosubmit:merge';
+      final IssueComment issueComment2 = IssueComment(
+        id: 2,
+        body: issueCommentBody2,
+        authorAssociation: 'OWNER',
+      );
+      commentList.add(issueComment);
+      commentList.add(issueComment2);
+      final MergeMethod mergeMethod = await validationService.checkForMergeMethod(
+        slug: slug,
+        prNumber: 1,
+      );
+      expect(MergeMethod.merge, mergeMethod);
+    });
+
+    test('Regex for merge method is case insensitive.', () async {
+      const String issueCommentBody = '@autosubmit:Rebase';
+      final IssueComment issueComment = IssueComment(
+        id: 1,
+        body: issueCommentBody,
+        authorAssociation: 'MEMBER',
+      );
+      commentList.add(issueComment);
+      final MergeMethod mergeMethod = await validationService.checkForMergeMethod(
+        slug: slug,
+        prNumber: 1,
+      );
+      expect(MergeMethod.rebase, mergeMethod);
+    });
+
+    test('Unknown merge method supplied returns merge as default.', () async {
```

Comment:
Your doctstring above states that "squash" is the default method, but here its merge? Maybe remove the specific strings, and let the default variable specify what the default should be?

Reply:
Good catch. Let me update this test description.

Reply:
Actually the test name is incorrect, because its actually squash thats default returned, please fix.